hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -168,6 +154,11 @@ public String name() {
     return name;
   }
 
+  @Override
+  public boolean tableExists(TableIdentifier identifier) {
+    return HiveCatalogUtil.isTableWithTypeExists(clients, identifier, TableType.EXTERNAL_TABLE);
```

Comment:
While it is a common Java convention to put "is" at the start of any method returning a boolean, we don't follow that in this project because it creates really awkward method names like this one. Instead, we use names that make sense when read, like `tableWithTypeExists` or `tableExistsWithType`.

Reply:
Is this an Iceberg requirement? It seems like this is more strict than we have been before. `HiveOperationsBase.validateTableIsIceberg` only requires `table_type` is `iceberg` (ignoring case).